src/tests/profiler/native/nongcheap/nongcheap.cpp

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "nongcheap.h"
+
+GUID NonGcHeapProfiler::GetClsid()
+{
+    // {EF0D191C-3FC7-4311-88AF-E474CBEB2859}
+    GUID clsid = { 0xef0d191c, 0x3fc7, 0x4311, { 0x88, 0xaf, 0xe4, 0x74, 0xcb, 0xeb, 0x28, 0x59 } };
+    return clsid;
+}
+
+HRESULT NonGcHeapProfiler::Initialize(IUnknown* pICorProfilerInfoUnk)
+{
+    Profiler::Initialize(pICorProfilerInfoUnk);
+
+    HRESULT hr = S_OK;
+    if (FAILED(hr = pCorProfilerInfo->SetEventMask2(
+        COR_PRF_ENABLE_OBJECT_ALLOCATED | COR_PRF_MONITOR_OBJECT_ALLOCATED,
```

Comment:
Should we have a specialized flag to capture NonGCHeap allocation like `COR_PRF_HIGH_MONITOR_LARGEOBJECT_ALLOCATED`?

Reply:
> Should we have a specialized flag

We discussed this and decided not to introduce it